src/Stratis.Bitcoin.Features.LightWallet/LightWalletSyncManager.cs

```diff
@@ -234,13 +235,13 @@ public void ProcessBlock(Block block)
                         token.ThrowIfCancellationRequested();
 
                         next = newTip.GetAncestor(next.Height + 1);
-                        Block nextblock = null;
+                        ChainedHeaderBlock nextblock = null;
                         int index = 0;
                         while (true)
                         {
                             token.ThrowIfCancellationRequested();
 
-                            nextblock = this.blockStore.GetBlockAsync(next.HashBlock).GetAwaiter().GetResult();
+                            nextblock = this.consensusManager.GetBlockDataAsync(next.HashBlock).GetAwaiter().GetResult();
```

Comment:
I think you need to also check `nextblock.Block == null`

Reply:
OK yeah fixed.